home4.1/CustomPizzaBuilder.java

```diff
@@ -0,0 +1,44 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+
+public class CustomPizzaBuilder {
+    private CustomPizza pizza;
+    private Map<String, Double> toppingPrices;
+
+    public CustomPizzaBuilder() {
+        this.pizza = new CustomPizza();
+        this.toppingPrices = new HashMap<>();
+        loadToppings("ingredients.txt"); // Load default toppings and prices from file
+    }
+
+    public void addTopping(String topping) {
+        if (toppingPrices.containsKey(topping)) {
+            pizza.addTopping(topping, toppingPrices.get(topping));
+        } else {
+            System.err.println("Topping not found: " + topping);
+        }
+    }
+
+    public CustomPizza getPizza() {
+        return pizza;
```

Comment:
pizza object should be reset somehow inside the Builder class to ensure that we can start building a new pizza after that call.